addon/convert-schema.js

```diff
@@ -0,0 +1,205 @@
+          return meetsCondition(dependencyValue, conditionValue)
+        })
+      })
+      if (hasDependencyMet && enableConditions.then !== undefined) {
+        props[key] = _.cloneDeep(enableConditions.then)
+      }
+      return hasDependencyMet
+    })
+  })
+  _.each(depsMet, function (dependencyMet, depName) {
+    const baseSchema = retModel.properties[depName]
+    if (dependencyMet && !baseSchema.set || !dependencyMet && baseSchema.set) {
+      retModel.properties[depName] = _.omit(_.defaults(props[depName] || {}, baseSchema), ['conditions', 'set'])
+    } else {
+      delete retModel.properties[depName]
+    }
+  })
+
+  return retModel
+}
+
+export default convertConditionalProperties
+
+function rowCellParents (container, rowIndex, rowCell, rowCellIndex) {
+  const subContainer = _.get(rowCell, 'container') || _.get(rowCell, 'item.container')
+
+  if (subContainer) {
+    this[subContainer] = {
+      container,
+      rowCellIndex,
+      rowIndex
+    }
+  }
+}
+
+function rowParents (container, row, index) {
+  _.each(row, rowCellParents.bind(this, container, index))
+}
+
+function findParentContainers (container, index) {
+  _.each(container.rows, rowParents.bind(this, container.id))
+}
+
+export function convertView (model, view) {
+  function getSchema (pathStack, model) {
+    if (pathStack.length <= 0 || model === undefined) {
+      return model
+    }
+
+    if (model.properties) {
+      const current = pathStack.pop()
+      return getSchema(pathStack, model.properties[current])
+    }
+
+    if (model.items) {
+      return getSchema(pathStack, model.items)
+    }
+  }
+
+  const parentContainers = {}
+
+  // Build list of each cells referencing
+  _.each(view.containers, findParentContainers.bind(parentContainers))
+
+  function getNextContainer (containerData) {
+    if (containerData) {
+      return _.find(view.containers, function (container) {
+        return container.id === containerData.container
+      })
+    }
+  }
+  function nextPathElement (containerData, nextContainer) {
+    if (containerData) {
+      return nextContainer.rows[containerData.rowIndex][containerData.rowCellIndex].model
+    }
+  }
+
+  function getFullPath (containerID) {
+    const path = []
+    let containerData = parentContainers[containerID]
+    let nextContainer = getNextContainer(containerData)
+    let pathElement = nextPathElement(containerData, nextContainer)
+    while (pathElement !== undefined) {
+      path.push(pathElement)
+      containerID = nextContainer.id
+      containerData = parentContainers[containerID]
+      nextContainer = getNextContainer(containerData)
+      pathElement = nextPathElement(containerData, nextContainer)
+    }
+    return path
+  }
+  let newView = _.cloneDeep(view)
+
+  // Filter out containers with non-existent properties
+  newView.containers = _.filter(_.map(view.containers, function (container, containerIndex) {
+    const newContainer = _.clone(container)
```

Comment:
The only major difference I can think of is that `\_.filter` works on objects/hashes in addition to arrays, but I have to admit I wasn't familiar with `\_.compact` so I might try to use that one in the future where applicable.